### PR TITLE
[CDAP-18450] Add support for configuring composite trigger properties

### PR DESCRIPTION
**Pull request by de-lan**
# Add support for configuring composite trigger properties

## Description
Add support for configuring composite trigger properties. When configuring the composite triggers, the runtime and plugin arguments can be individually configured within the triggering pipeline. The compute file can be configured in the top layer of the page. The sister backend PR is: https://github.com/cdapio/cdap/pull/14176

## PR Type
- [ ] Bug Fix
- [x] Feature
- [ ] Build Fix
- [ ] Testing
- [ ] General Improvement
- [ ] Cherry Pick

## Links
Jira: [CDAP-18450](https://cdap.atlassian.net/browse/CDAP-18450)

## Test Plan
Manual + E2E

## Screenshots
image
image
image

### CODE REVIEW

**Comment by seanfeld** [comment]
is this all new code?

File: app/cdap/components/PipelineTriggers/store/PipelineTriggersActionCreator.ts [hunk 1]
```diff
@@ -381,57 +408,33 @@ export function generateRuntimeMapping(mapping) {
...
-  mapping: any,
-  propertiesConfig: any,
-  triggersGroupRunArgsToAdd: ITriggersGroupRunArgs,
-  selectedNamespace: string,
-  triggeringPipelineInfo: ITriggeringPipelineInfo
-) {
-  const triggersGroupRunArgs = { ...triggersGroupRunArgsToAdd };
-  mapping.forEach((map) => {
-    const keySplit = map.key.split(':');
+export function validateTriggerMappping(
```
Comment (seanfeld) [inline app/cdap/components/PipelineTriggers/store/PipelineTriggersActionCreator.ts:417]: so this function returns null if the result is valid and it returns a string if it is invalid?

File: app/cdap/components/PipelineTriggers/store/PipelineTriggersActionCreator.ts [hunk 2]
```diff
@@ -341,10 +339,39 @@ export function addToTriggerGroup(
     },
   ];
 
+  const triggersGroupRunArgs = { ...triggersGroupRunArgsToAdd };
+
+  if (!!mapping && !!triggersGroupRunArgs) {
+    mapping.forEach((map) => {
+      const keySplit = map.key.split(':');
+      const currpPipelineId = {
```
Comment (seanfeld) [inline app/cdap/components/PipelineTriggers/store/PipelineTriggersActionCreator.ts]: currp?

File: app/cdap/components/PipelineTriggers/store/PipelineTriggersActionCreator.ts (hunk 1 above)
Comment (seanfeld) [reply app/cdap/components/PipelineTriggers/store/PipelineTriggersActionCreator.ts:417]: hmm okay i guess it tells you what it returns in the js doc - can you move a summary of that explanation to the top part of the description as well?

**Review by seanfeld — APPROVED** [review]
Just one nit

**Comment by de-lan** [comment]
> is this all new code?


Mostly refactoring the UI and bug fixes

File: app/cdap/components/PipelineTriggers/PayloadConfigModal/index.js [hunk 3]
```diff
@@ -33,41 +32,24 @@ export default class PayloadConfigModal extends Component {
...
-    onEnableSchedule: PropTypes.func,
+    onConfigureSchedule: PropTypes.func,
     disabled: PropTypes.bool,
     scheduleInfo: PropTypes.object,
     configureError: PropTypes.string,
     onToggle: PropTypes.func,
-    andTriggersEnabled: PropTypes.bool,
-  };
-
-  state = {
```
Comment (de-lan) [inline app/cdap/components/PipelineTriggers/PayloadConfigModal/index.js:45]: Moved out the payloadConfig state and the internal button. The original code brings the issue that the payload modal button is not easily adjustable.

File: app/cdap/components/PipelineTriggers/PayloadConfigModal/index.js [hunk 4]
```diff
@@ -103,22 +86,6 @@ export default class PayloadConfigModal extends Component {
...
 
   render() {
-    const label = this.props.disabled
-      ? T.translate(`${PREFIX}.configPayloadBtnDisabled`)
-      : T.translate(`${PREFIX}.configPayloadBtn`);
-    const btnCy = this.props.disabled ? 'view-payload-btn' : 'trigger-config-btn';
-
-    return (
-      <div className="payload-config-modal">
-        <Button
```
Comment (de-lan) [inline app/cdap/components/PipelineTriggers/PayloadConfigModal/index.js:113]: The button is now moved to the parent component for flexibility.

File: app/cdap/components/PipelineTriggers/store/PipelineTriggersActionCreator.ts (hunk 2 above)
Comment (de-lan) [reply app/cdap/components/PipelineTriggers/store/PipelineTriggersActionCreator.ts]: ooops, typo.... will rename to `currentPipelineId `

File: app/cdap/components/PipelineTriggers/store/PipelineTriggersActionCreator.ts (hunk 1 above)
Comment (de-lan) [reply app/cdap/components/PipelineTriggers/store/PipelineTriggersActionCreator.ts:417]: yep ur right, updated

**Pull request merged**